exp/utils.py

```diff
@@ -0,0 +1,62 @@
+import csv
+import io
+
+
+def flatten_dict(d):
+    """Flatten a dictionary where values may be other dictionaries
+
+	The dictionary returned will have keys created by joining higher- to lower-level keys with dots. e.g. if the original dict d is
+	{'a': {'x':3, 'y':4}, 'b':{'z':5}, 'c':{} }
+	then the dict returned will be
+	{'a.x':3, 'a.y': 4, 'b.z':5}
+
+	Note that if a key is mapped to an empty dict or list, NO key in the returned dict is created for this key.
+
+	Also note that values may be overwritten if there is conflicting dot notation in the input dictionary, e.g. {'a': {'x': 3}, 'a.x': 4}.
+	"""
+    # http://codereview.stackexchange.com/a/21035
+
+    def expand(key, value):
+        if isinstance(value, list):
+            value = {i: v for (i, v) in enumerate(value)}
+        if isinstance(value, dict):
+            return [
+                (str(key) + "." + str(k), v) for k, v in flatten_dict(value).items()
+            ]
+        else:
+            return [(key, value)]
+
+    items = [item for k, v in d.items() for item in expand(k, v)]
+
+    return dict(items)
+
+
+def csv_dict_output_and_writer(headerList):
```

Comment:
does black not change `headerList` to `header_list`? We may need to bring pyflakes as a dependency to make sure we adhere to PEP8

Reply:
sorry about that - no, black does not change the variable names, I do run it!

Reply:
argh yeah ok after this PR I'll plug in a dependency and add an `invoke` script to run all the housekeeping goodies. I've been too reliant on my own PyCharm setup

Reply:
ooh thanks! my apologies, camelcasing variables got drilled into me very early and has been hard to kick